numpy/core/src/umath/ufunc_object.c

```diff
@@ -2343,6 +2344,16 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
         }
     }
 
+    total_problem_size = NpyIter_GetIterSize(iter);
+    if (total_problem_size < 0) {
```

Comment:
If it would not care explicitly, it should error out. So it sets it to 
-1 and errors out unless you remove that stuff again. I added that
weird feature explicitly for these gufuncs that bloat the iterator.